resolvers.py

```diff
@@ -0,0 +1,1979 @@
+
+        qs = PeopleOnBenchHelper().add_capability_filter(capability_name, band_name)
+        return OverallFreePoolHelper(qs).redirect_columns(column)
+
+
+class LastApprovedResolver:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        return core_models.LastApproved.objects.last()
+
+
+class UtilizationSnapshotReportResolver:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+
+        # get current period name
+        current_period = models.Periods.objects.filter(
+            period_tag="current_period").values()[0]
+
+        current_period_name = current_period["period_name"]
+        current_period_start_date = current_period["period_start_date"]
+
+        qs = core_models.UtilizationFactBackup.objects.all()
+
+        qs_result = {}
+
+        # going over all the data of backup
+        for val in qs.values():
+            category = "Previous Period"
+
+            if current_period_name == val["period_name"]:
+
+                # to check if data is of current period
+                if val["start_date"] < current_period_start_date:
+                    continue
+                category = "Current Period"
+
+            # going over snapshot data of every row in backup table.
+            for snap_data in json.loads(val["snapshot_data"]):
+                # key to seperate different billing attribute and billing attribute type
+                key = snap_data["billing_attribute"] + \
+                    "_" + snap_data["billing_attribute_type"]
+
+                data_value = {
+                    "period_name": val["period_name"],
+                    "week_name": val["week_name"],
+                    "start_date": str(val["start_date"]),
+                    "end_date": str(val["end_date"]),
+                    "record_type": val["record_type"],
+                    "category": category,
+                    "billing_attribute": snap_data["billing_attribute"],
+                    "billing_attribute_type": snap_data["billing_attribute_type"],
+                    "fte_count": snap_data["fte_count"],
+                    "fte_percentage": snap_data["fte_percent"]
+                }
+
+                if key not in qs_result:
+                    qs_result[key] = {"billing_attribute": snap_data["billing_attribute"],
+                                      "billing_attribute_type": snap_data["billing_attribute_type"], "value": []}
+
+                qs_result[key]["value"].append(data_value)
+
+        final_result = []
+
+        data_id = 1
+        # adding value to list of objects
+        for data in qs_result.values():
+            data["id"] = data_id
+            data_id = data_id + 1
+            final_result.append(data)
+
+        return final_result
+
+
+class UserManualResolver:
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        report_name = kwargs.get("report_name")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

You are using the `get` method without a default argument to return the value of a key in a dictionary. We recommended that you use a default argument so that if the value for your key is not found, a default value is returned. If a default value is not provided and the key is not found, then `None` is returned.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)